parachain/pallets/omni-account/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub(crate) mod identity_context;
+
+pub use core_primitives::{assertion::network::Web3Network, Identity, OmniAccountIdentity};
+pub use frame_system::pallet_prelude::BlockNumberFor;
+pub use pallet::*;
+
+use frame_support::pallet_prelude::*;
+use frame_system::pallet_prelude::*;
+use identity_context::*;
+use sp_core::H256;
+use sp_core_hashing::blake2_256;
+use sp_std::vec::Vec;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	/// The current storage version.
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	#[pallet::pallet]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The event type of this pallet.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		/// The origin which can manage the pallet.
+		type TEECallOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		/// The maximum number of identities an id graph can have.
+		#[pallet::constant]
+		type MaxIDGraphLength: Get<u32>;
+	}
+	pub type IDGraphLinks<T> =
+		BoundedVec<(OmniAccountIdentity, IdentityContext<T>), <T as Config>::MaxIDGraphLength>;
+
+	#[pallet::storage]
+	pub type LinkedIdentities<T: Config> =
+		StorageMap<Hasher = Blake2_128Concat, Key = H256, Value = ()>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn id_graphs)]
+	pub type IDGraphs<T: Config> =
+		StorageMap<Hasher = Blake2_128Concat, Key = Identity, Value = IDGraphLinks<T>>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Identity linked
+		IdentityLinked { who: Identity, identity: H256 },
+		/// Identity deactivate_identity
+		IdentityDeactivated { who: Identity, identity: H256 },
+		/// Identity activate_identity
+		IdentityActivated { who: Identity, identity: H256 },
+		/// Web3 networks updated
+		Web3NetworksUpdated { identity: H256, web3networks: Vec<Web3Network> },
+		/// Identity remove
+		IdentityRemoved { who: Identity, identities: Vec<OmniAccountIdentity> },
+		/// Identity made public
+		IdentityMadePublic { who: Identity, identity: H256 },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The Identity is already linked
+		IdentityAlreadyLinked,
+		/// IDGraph len limit reached
+		IDGraphLenLimitReached,
+		/// Identity not found
+		IdentityNotFound,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal))]
+		pub fn link_identity(
+			origin: OriginFor<T>,
+			who: Identity,
+			identity: OmniAccountIdentity,
+			web3networks: Vec<Web3Network>,
+		) -> DispatchResult {
+			let _ = T::TEECallOrigin::ensure_origin(origin)?;
+			let identity_hash = H256::from(blake2_256(&identity.encode()));
```

Comment:
If we ever decide to introduce significant change to `OmniAccountIdentity` or `Identity` it will require storage migration as encoding is coupled to data structure. 

What do you think of using hash of DID instead?

Reply:
Yeah I think it's a good idea 👍🏼